src/java.rmi/share/classes/sun/rmi/runtime/NewThreadAction.java

```diff
@@ -48,34 +46,26 @@
  *
  * @author      Peter Jones
  **/
-public final class NewThreadAction implements PrivilegedAction<Thread> {
+public final class NewThreadAction {
```

Comment:
Suggest inlining the methods and getting rid of the class, moving non-trivial methods to an existing Util class.
Or at least renaming it to to drop the "Action" that is indicative of doPrivileged.

Reply:
Yes, further cleanup should be done here. I had started on it but I backed away since it felt like a distraction from the initial pass of removing direct SecurityManager and AccessController calls. I've filed [JDK-8344461](https://bugs.openjdk.org/browse/JDK-8344461) to cover the followup work.